src/index.ts

```diff
@@ -58,20 +59,35 @@ async function main() {
         workflow_id,
         branch,
       });
+      console.log(`Found ${data.total_count} runs total.`);
 
-      const branchWorkflows = data.workflow_runs.filter(run => run.head_branch === branch);
-      console.log(`Found ${branchWorkflows.length} runs for workflow ${workflow_id} on branch ${branch}`);
-      console.log(branchWorkflows.map(run => `- ${run.html_url}`).join('\n'));
+      let cancel_before;
+      if (all_but_latest) {
+        cancel_before = new Date(data.workflow_runs.reduce((a, b) => Math.max(a.created_at, b.created_at)));
+      } else {
+        cancel_before = new Date(current_run.created_at);
+      }
 
-      const runningWorkflows = branchWorkflows.filter(run =>
-        (ignore_sha || run.head_sha !== headSha) &&
-        run.status !== 'completed' &&
-        new Date(run.created_at) < new Date(current_run.created_at)
+      const runningWorkflows = data.workflow_runs.filter(
+        run => run.head_branch === branch && (ignore_sha || run.head_sha !== headSha) && run.status !== 'completed' &&
+            run != current_run &&
+            new Date(run.created_at) < cancel_before
       );
-      console.log(`with ${runningWorkflows.length} runs to cancel.`);
-
-      for (const {id, head_sha, status, html_url} of runningWorkflows) {
-        console.log('Canceling run: ', {id, head_sha, status, html_url});
+      console.log(`Found ${runningWorkflows.length} runs to cancel.`);
+      for (const {id, head_sha, status} of runningWorkflows) {
+        console.log('Cancelling another run: ', {id, head_sha, status});
+        const res = await octokit.actions.cancelWorkflowRun({
+          owner,
+          repo,
+          run_id: id
+        });
+        console.log(`Cancel run ${id} responded with status ${res.status}`);
+      }
+      // Make sure we cancel this run itself if it's out-of-date.
+      // We postponed canceling this run because otherwise we couldn't cancel the rest.
+      if (all_but_latest && new Date(current_run.created_at) < cancel_before) {
+        // FIXME actions/core doesn't support cancelling, so we need to do it through the API
```

Comment:
I think you can remove the FIXME comment since there is nothing to fix in my opinion

Reply:
I think the fixme was legacy from the original version on top of which this was implemented. It's removed now.